storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -500,22 +500,29 @@ public static void validateBrokerLogConfigValues(Map<?, ?> props,
      * The default values should be extracted from the KafkaConfig.
      * @param props The properties to be validated
      */
-    private static void validateTopicLogConfigValues(Map<?, ?> props,
-                                                     boolean isRemoteLogStorageSystemEnabled) {
+    public static void validateTopicLogConfigValues(Map<?, ?> props,
```

Comment:
If we are planning to validate only the case where tiered storage functionality is enabled (or) disabled in the broker during restarts. Can we use the `validateRemoteStorageOnlyIfSystemEnabled` instead of `validateTopicLogConfigValues`?

Reply:
I am ambivalent. If I am to use `validateRemoteStorageOnlyIfSystemEnabled` I would still need another validation function to check for the topic-level config, no? Otherwise the `validateRemoteStorageOnlyIfSystemEnabled` will only operate based on knowledge of the system-wide config. Are you okay with this?

Reply:
I agree with @kamalcph  that we actually doesn't want to validate anything else except `validateRemoteStorageOnlyIfSystemEnabled`.

Reply:
Okay, hopefully addressed in the latest commit